Sources/dump-unified-graph/DumpUnifiedGraph.swift

```diff
@@ -0,0 +1,109 @@
+          help: "whether to pretty-print the output JSON (default: true)")
+    var prettyPrint: Bool = true
+
+    @Option(name: .short, help: "output file to write to (default: standard out)")
+    var output: String?
+
+    @Option(help: "directory to recursively load symbol graphs from", completion: .directory)
+    var symbolGraphDir: String?
+
+    @Argument(help: "list of symbol graphs to load", completion: .file(extensions: ["json"]))
+    var files: [String] = []
+
+    mutating func validate() throws {
+        guard !files.isEmpty || symbolGraphDir != nil else {
+            throw ValidationError("Please provide files or a symbol graph directory")
+        }
+
+        if let symbolGraphDir = symbolGraphDir, !symbolGraphDir.hasSuffix("/") {
+            self.symbolGraphDir = symbolGraphDir.appending("/")
+        }
+    }
+
+    func run() throws {
+        var symbolGraphs = files
+        if let symbolGraphDir = symbolGraphDir {
+            symbolGraphs.append(contentsOf: loadSymbolGraphsFromDir(symbolGraphDir))
+        }
+
+        if symbolGraphs.isEmpty {
+            print("error: No symbol graphs were available")
+            throw ExitCode.failure
+        }
+
+        let decoder = JSONDecoder()
+        let collector = GraphCollector()
+
+        for symbolGraph in symbolGraphs {
+            let graphUrl = URL(fileURLWithPath: symbolGraph)
+            let decodedGraph = try decoder.decode(SymbolGraph.self, from: Data(contentsOf: graphUrl))
+            collector.mergeSymbolGraph(decodedGraph, at: graphUrl)
+        }
+
+        let (unifiedGraphs, _) = collector.finishLoading()
+        let outputGraph: UnifiedSymbolGraph
+
+        if let moduleName = moduleName {
+            if let graph = unifiedGraphs[moduleName] {
+                outputGraph = graph
+            } else {
+                print("error: The given module was not represented in the symbol graphs")
+                throw ExitCode.failure
+            }
+        } else {
+            if unifiedGraphs.count > 1 {
+                print("error: No module was given, but more than one module was represented in the symbol graphs")
+                throw ExitCode.failure
+            } else {
+                outputGraph = unifiedGraphs.values.first!
+            }
+        }
+
+        let encoder = JSONEncoder()
+        if prettyPrint {
+            encoder.outputFormatting.insert(.prettyPrinted)
+        }
+
+        let encoded = try encoder.encode(outputGraph)
+
+        if let output = output, output != "-" {
+            FileManager.default.createFile(atPath: output, contents: encoded)
+        } else {
+            let outString = String(data: encoded, encoding: .utf8)
+            print(outString!)
+        }
+    }
+}
+
+func loadSymbolGraphsFromDir(_ dir: String) -> [String] {
+    let enumerator = FileManager.default.enumerator(atPath: dir)
+    var symbolGraphs: [String] = []
+
+    while let filename = enumerator?.nextObject() as? String {
```

Comment:
```suggestion
    for case let filename as String in enumerator! {
```

Reply:
I'm going off the docs for `FileManager.enumerator()` 🙃 but this looks much nicer, thanks!

Reply:
One issue i would have with this is that if `enumerator()` returns nil, this will crash, whereas in the current code it will return no symbol graphs and print an error. The docs don't say _when_ `FileManager.enumerator(atPath:)` will return nil, so i can only guess, but i'd rather guard against it than assume.

Reply:
Yeah when I tried this out locally, I also modified line 127 to:
```swift
guard let enumerator = FileManager.default.enumerator(atPath: dir) else {
    return []
}
```